model/fn-execution/src/main/proto/beam_fn_api.proto

```diff
@@ -266,6 +266,40 @@ message Metrics {
       map<string, int64> output_elements_remaining = 3;
     }
 
+    // User defined metrics
+    message User {
+
+      // A key for identifying a metric at the most granular level.
+      message MetricKey {
```

Comment:
We should probably be calling this a MetricName to avoid confusion with the MetricKey class in Java/Python that contains a step id.